xarray/backends/netCDF4_.py

```diff
@@ -68,7 +68,7 @@ def __init__(self, variable_name, datastore):
             # use object dtype because that's the only way in numpy to
             # represent variable length strings; it also prevents automatic
             # string concatenation via conventions.decode_cf_variable
-            dtype = np.dtype("O")
+            dtype = coding.strings.create_vlen_dtype(str)
```

Comment:
This change adds the string metadata to the object dtype. It should just give us the ability, to check on the dtype within the object in further use.

Reply:
Can we update the comment above to to align with this change. Its not clear from reading this if the dtype is still `O`.